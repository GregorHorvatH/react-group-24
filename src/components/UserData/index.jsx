const UserData = ({ user }) => {
  const { name, dob, email, picture } = user;

  return name?.first ? (
    <div className="user-data">
      <img src={picture?.large} alt={name?.first} />
      <p>FirstName: {name?.first}</p>
      <p data-testid="lastname">LastName: {name?.last}</p>
      <p>Email: {email}</p>
      <p>Age: {dob?.age}</p>
    </div>
  ) : (
    <p>Loading...</p>
  );
};

export default UserData;
