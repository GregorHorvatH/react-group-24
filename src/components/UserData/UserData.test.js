import { render } from '@testing-library/react';
import UserData from './';

const user = {
  firstName: 'Bobbovych',
  lastName: 'Bobby',
  email: 'bobby@gmail.com',
  age: 15,
};

describe('UserData', () => {
  test('should not blow up', () => {
    const { container } = render(<UserData user={user} />);

    expect(container).toMatchSnapshot();
  });

  test('should have 4 ps', () => {
    const { container } = render(<UserData user={user} />);

    expect(container.querySelectorAll('p').length).toEqual(1);
  });

  // test('should have Bobby as lastname', () => {
  //   const { getByTestId } = render(<UserData user={user} />);

  //   expect(getByTestId('lastname').textContent).toEqual('LastName: Bobby');
  // });
});

// import { sum } from './utils';

// describe('sum should work correctly', () => {
//   test('(1+1)', () => {
//     const result = sum(1, 1);

//     expect(result).toEqual(2);
//   });

//   test('(2+5)', () => {
//     expect(sum(2, 5)).toEqual(7);
//   });

//   test('(-10+15)', () => {
//     expect(sum(-10, 15)).toEqual(5);
//   });
// });
