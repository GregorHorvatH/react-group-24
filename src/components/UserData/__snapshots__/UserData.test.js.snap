// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserData should not blow up 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
